### DISCUSSION TITLE
Two Credit Card-related Features

### DISCUSSION THREAD

**Discussion by Sanhom365** [firefly-iii/firefly-iii#7164] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)
- [X] My request is not listed as [a very good idea, but unfortunately...](https://docs.firefly-iii.org/firefly-iii/about-firefly-iii/what-its-not/)
- [X] I've used [the search](https://github.com/firefly-iii/firefly-iii/issues?q=is%3Aissue) and this has not been discussed before.

### Description

Thank you very much for the Firefly III software! The user experience is really great, and I hope you will continue to improve!  
While using the software, I found some areas that do not meet practical needs.

-  I would like to suggest adding a "Discount" option within the "Expenses" and "Transfers" items in Firefly III.  
-  I would also like to suggest adding a billing period for credit cards in Firefly III.

`Thank you!`

### Solution

- For example, if I have a 10 yuan coupon on the platform for a credit card repayment of 1000 yuan, I would like to transfer only 990 yuan to the credit card and apply the coupon for the repayment.
- For example, if the billing date is on the 15th of every month, the billing cycle could be set from the 16th of the current month to the 15th of the following month.

**Comment by JC5** [comment]
1. No. If you want to add a discount to what you buy, you'll have to subtract the discount yourself. Firefly III won't manage the discount, just the payments.
2. There is a billing date feature for credit cards but it will not be automated: https://github.com/firefly-iii/firefly-iii/issues/6774

**Reply by waicool20** [reply]
If I want to itemize a transaction, I usually have to pick an item and subtract the discount/gift value hiding its original value. Would be nice to have a field that allows zero or negative values that offset up to the entire total of the transaction.

**Comment by Sanhom365** [comment]
Thank you for your reply.

1. Does that mean when I use 990 yuan from another account to pay off 1000 yuan credit card balance, and use a voucher to offset the remaining 10 yuan, I can only use income to pay off the 10 yuan because it did not come from my original account but was given to me as a gift by the credit card issuer?
2. Alright, I'll look into it some more.

**Comment by JC5** [comment]
1. Yes, exactly :)

**Reply by geomule** [reply]
Ok, I came up with similar problem where I was given a 50 euro Amazon gift coupon and used it to pay 60 euro of purchase. Effectively, only 10 euro was paid from credit card. Now, if split accounts on the source were possible, I would have accounted for the coupon as "cash" and sent it from "presents" (revenue) to "cash wallet". Then I would have split the source as 50 euro on "cash wallet" and 10 euro on "credit card" with all 60 euro going to "Electronics". As split accounts are not possible on the source from what I gather, how else can I solve this one? Any particular reason why split purchases on the source are not contemplated in Firefly iii and are there any plans to implement in the future?
